switch/libsmb2/libsmb2-91d6c8a-fix_getaddrinfo_hints.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/lib/socket.c b/lib/socket.c
+index 978f06d..156fad5 100644
+--- a/lib/socket.c
++++ b/lib/socket.c
+@@ -701,9 +747,14 @@ smb2_connect_async(struct smb2_context *smb2, const char *server,
+         } else {
+                 port = "445";
+         }
++        
++        struct addrinfo hints;
++        memset(&hints, 0, sizeof hints);
++        hints.ai_family = AF_INET; // AF_INET or AF_INET6 to force version
++        hints.ai_socktype = SOCK_STREAM;
```

Comment:
Was always throwing error 22 without this. Found an existing issue reporting the same problem, copy/pasted solution. Probably a better way to fix this though...

Reply:
Fwiw getaddrinfo with hints=NULL is currently broken. However, a libnx branch I'm working on (https://github.com/switchbrew/libnx/tree/socket-cleanup) fixes this.

Reply:
thanks, I'll remove this patch once fixed in libnx then.